### ISSUE TITLE
Test harness and spec

### ISSUE THREAD

**Issue by gedw99** [livghit/go-gui-htmx#1]
Hey @livghit 

I have a basic spec and am working on a test harness.

Shall I add this to the repo ?

**Comment by livghit** [comment]
Yep every additional info is welcomed . We could harvest as much information and ideas as possible so we can shape a project out of this . What do you think ?

**Comment by gedw99** [comment]
Yep that’s the idea :) 

My appproach is decent but it’s still a fair bit of work to get a basic version working.

I tend to use make files to speed up things. Hope that’s ok ?

**Comment by livghit** [comment]
Yup is perfect , makefiles are very useful 👍

**Comment by joeblew99** [comment]
I created the SPEC and TODO to flesh out the basic high level approach that i think would work.

also i did some research on goalng htmx frameworks and https://github.com/kyoto-framework/kyoto might be a good one.

i was looking fro one where the client is very simple, because we will have to do the same in golang frontend.
https://github.com/kyoto-framework/kyoto/blob/master/client/src/client.ts is very simple and is based on alpine morph.

https://kyoto.codes/kyoto/actions.html shows that the markup action language uses notation of "ssa" which i am guessing is "server side action", which is exactly what we are doing.

**Comment by livghit** [comment]
Had the chance to take a deeper look into the kyoto framework and I musst admit I kinda like it . Its something like Fiber but way better for our purpouse . I will play a bit with it today to understand it but in general like I saw its preatty easy and understandable . What do you think about wasm should we implement a wrapper around it too so we can make working with wasm easier in case we or someone else needs it ?

**Comment by gedw99** [comment]
You can grab me here 👍 
https://t.me/gedw99

**Comment by livghit** [comment]
Sure Liviu_g04@yahoo.com
